packages/blobs/server.js

```diff
@@ -99,19 +107,27 @@ Blobs2.delete = function (selector, callback) {
   } else {
     check(selector, Object);
   }
-  if (typeof callback !== "function") throw new Meteor.Error("no-callback");
+  if (callback && typeof callback !== "function") {
+    throw new Meteor.Error("invalid-callback");
+  }
 
   var rmPromises = [];
   Blobs2.find(selector).forEach(function (blob) {
     rmPromises.push(Q.nfcall(remove, blob.getFilePath()));
   });
 
-  Q.all(rmPromises)
+  // NOTE: anything attached to the returned promise will happen *after*
+  // the .then and .catch here.
+  return Q.all(rmPromises)
     .then(Meteor.bindEnvironment(function (result) {
-      callback(null, rmPromises.length);
+      var asdf = Blobs2.remove(selector, { multi: true });
```

Comment:
this poorly-named variable never gets used... Also yop below, another placeholder name